java/vector/src/main/java/org/apache/arrow/vector/BaseNullableVariableWidthVector.java

```diff
@@ -836,17 +858,259 @@ public int getLastSet() {
     * @return starting offset for the element
     */
    public long getStartEnd(int index) {
-      return (long)offsetBuffer.getInt(index * OFFSET_WIDTH);
+      return offsetBuffer.getLong(index * OFFSET_WIDTH);
    }
 
    /**
     * Mark the particular position in the vector as non-null.
     *
     * @param index position of the element.
     */
+   @Override
    public void setIndexDefined(int index) {
```

Comment:
It seems naming conversion wise, all methods that handles allocates new buffers are called "xxxSafe". Does
```
public void setValiditySafe(int index, boolean valid)
```

Makes more sense for this function?

Reply:
Not quite sure. Since this has always been setIndexDefined(), I just left it as is

Reply:
Ok